addons/web/static/src/search/search_model.js

```diff
@@ -960,13 +954,26 @@ export class SearchModel extends EventBus {
                     );
                 }
             } else {
+                if (generatorId.startsWith("custom")) {
+                    this.query = this.query.filter(
+                        (queryElem) => queryElem.searchItemId !== searchItemId
+                    );
+                    this.query.push({ searchItemId, generatorId });
+                    continue;
+                }
+                this.query = this.query.filter(
+                    (queryElem) =>
+                        queryElem.searchItemId !== searchItemId ||
+                        !queryElem.generatorId.startsWith("custom")
+                );
```

Comment:
I am not sure, but I have the feeling the filter:
```js
                    this.query = this.query.filter(
                        (queryElem) => queryElem.searchItemId !== searchItemId
                    );
```
is included in the second one and so I am wondering if it is not equivalent to that: 

```suggestion
                this.query = this.query.filter(
                    (queryElem) =>
                        queryElem.searchItemId !== searchItemId ||
                        !queryElem.generatorId.startsWith("custom")
                );
```
🤔 

What do you think? 🧐

Reply:
This works, except we don't want the code below this comment to run if we are selecting a custom option, but it can be simplified for sure

Reply:
Actually, now that I think about it more, this wouldn't work, because the filters keep the elements that match the condition, meaning that if we extend the condition of the first filter to use the condition of the second, we'll keep too many options enabled (we won't disable non custom options when selecting a custom option)

Reply:
Ah okay, I see, thanks for the explanation. 🙂